pkg/util/pem.go

```diff
@@ -120,3 +125,19 @@ func DecodeX509CertificateChainBytes(certBytes []byte) ([]*x509.Certificate, err
 
 	return certs, nil
 }
+
+// Check if the given certificate was added to certificates bundle already
+func IsCertificateDuplicate(certsHash map[[32]byte]struct{}, certData []byte) bool {
+	// calculate hash sum of the given certificate
+	hash := sha256.Sum256(certData)
```

Comment:
I note that `crypto.CertPool` uses `sha256.Sum224` for deduplicating certs. Any particular reason you went for something else?

Reply:
Interesting that CertPool uses sha224! I think sha256 is fine here though, it's what I'd have reached for as a "default"

Reply:
Hello,
Thank you so much for your feedback, it is very appreciated!
There weren't any reasons to use sha224 against sha256, they have almost equal computation complexity, bus some differences have present:

https://www.rfc-editor.org/rfc/rfc3874.txt

> 1.1.  Usage Considerations
> 
>    Since SHA-224 is based on SHA-256, roughly the same amount of effort
>    is consumed to compute a SHA-224 or a SHA-256 digest message digest
>    value.  Even though SHA-224 and SHA-256 have roughly equivalent
>    computational complexity, SHA-224 is an appropriate choice for a
>    one-way hash function that provides 112 bits of security.  The use of
>    a different initial value ensures that a truncated SHA-256 message
>    digest value cannot be mistaken for a SHA-224 message digest value
>    computed on the same data.
> 
>    Some usage environments are sensitive to every octet that is
>    transmitted.  In these cases, the smaller (by 4 octets) message
>    digest value provided by SHA-224 is important.
> 
>    These observations lead to the following guidance:
> 
>    *  When selecting a suite of cryptographic algorithms that all offer
>       112 bits of security strength, SHA-224 is an appropriate choice
>       for one-way hash function.
> 
>    *  When terseness is not a selection criteria, the use of SHA-256 is
>       a preferred alternative to SHA-224.

Reply:
For our purposes we have to choice algorithm which proposes:
- acceptable computation complexity and
- minimize hash collisions

Reply:
I'm fine with sha256. 😃